pkg/server/batch_check.go

```diff
@@ -87,6 +126,21 @@ func (s *Server) BatchCheck(ctx context.Context, req *openfgav1.BatchCheckReques
 		methodName,
 	).Observe(queryCount)
 
+	for _, item := range result {
+		batchCheckCheckDurationHistogram.WithLabelValues(
+			strconv.FormatUint(uint64(item.DatastoreQueryCount), 10), // TODO bucketize
+			req.GetConsistency().String(),
+		).Observe(float64(item.Duration.Milliseconds()))
+
+		hadError := "false"
+		if item.Err != nil {
+			hadError = "true"
+		}
+
+		// TODO make this specifically 500s, not other errors
+		batchCheckResultCounter.WithLabelValues(hadError).Inc()
```

Comment:
this gets roughly what we want in terms of error rate:
<img width="807" alt="image" src="https://github.com/user-attachments/assets/521b4234-19e4-48fe-a2c9-3b25c9d096ab">